crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -621,4 +661,56 @@ mod test {
         // But, even after the sync phase, internal alarms are not converted and just ignored, as they are purely internal
         assert!(converter.convert(&internal_alarm_message).is_empty());
     }
+
+    #[test]
+    fn test_convert_event() -> Result<()> {
+        let size_threshold = SizeThreshold(32 * 1024);
+        let device_name = String::from("test");
+        let device_type = String::from("test_type");
+        let http_proxy = Box::new(MockC8YHttpProxy::new());
+
+        let mut converter =
+            CumulocityConverter::new(size_threshold, device_name, device_type, http_proxy);
+
+        let event_topic = "tedge/events/click_event";
+        let event_payload = r#"{ "message": "Someone clicked" }"#;
+        let event_message = Message::new(&Topic::new_unchecked(event_topic), event_payload);
+
+        let converted_events = converter.convert(&event_message);
+        assert_eq!(converted_events.len(), 1);
+        let converted_event = converted_events.get(0).unwrap();
+        assert_eq!(converted_event.topic.name, "c8y/s/us");
+        assert!(converted_event
+            .payload_str()?
+            .starts_with("400,click_event"));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_convert_big_event() {
+        let size_threshold = SizeThreshold(32 * 1024);
+        let device_name = String::from("test");
+        let device_type = String::from("test_type");
+
+        let mut http_proxy = Box::new(MockC8YHttpProxy::new());
+        http_proxy
+            .expect_send_event()
+            .with(
+                predicate::eq("click_event"),
+                predicate::always(),
+                predicate::always(),
+            )
+            .returning(|_, _, _| Ok("123".into()));
+
+        let mut converter =
+            CumulocityConverter::new(size_threshold, device_name, device_type, http_proxy);
+
+        let event_topic = "tedge/events/click_event";
+        let big_event_message = create_packet(1024 * 20); // Event payload > 16K
+        let big_event_payload = json!({ "message": big_event_message }).to_string();
+        let big_event_message = Message::new(&Topic::new_unchecked(event_topic), big_event_payload);
+
+        assert!(converter.convert(&big_event_message).is_empty());
+    }
```

Comment:
I like this test.

I was wondering if we could have an integration test with a mock http server to run something similar on non fake implementation?

Reply:
Definitely. I'd still keep this test as-is. But, the unit test for `JwtAuthHttpProxy::send_event()` could be written like that. Will give it a try.